NLog.Web.AspNetCore/LayoutRenderers/AspNetRequestIpLayoutRenderer.cs

```diff
@@ -16,18 +19,59 @@ namespace NLog.Web.LayoutRenderers
     [LayoutRenderer("aspnet-request-ip")]
     public class AspNetRequestIpLayoutRenderer : AspNetLayoutRendererBase
     {
+        private const string ForwardedForHeader = "X-Forwarded-For";
+
+        /// <summary>
+        /// Gets or sets whether the renderer should check value of X-Forwarded-For header
+        /// </summary>
+        /// <docgen category='Rendering Options' order='10' />
+        public bool CheckForwardedForHeader { get; set; }
+
         /// <summary>
         /// Render IP
         /// </summary>
         protected override void DoAppend(StringBuilder builder, LogEventInfo logEvent)
         {
+            var forwardedIp = string.Empty;
             var httpContext = HttpContextAccessor.HttpContext;
+
 #if !ASP_NET_CORE
-            var ip = httpContext.TryGetRequest()?.ServerVariables["REMOTE_ADDR"];
+            var request = httpContext.TryGetRequest();
+
+            if (request == null)
+            {
+                return;
+            }
+
+            if (CheckForwardedForHeader)
+            {
+                var forwardedHeader = request.Headers[ForwardedForHeader];
+
+                if (!string.IsNullOrEmpty(forwardedHeader))
+                {
+                    var addresses = forwardedHeader.Split(',');
+                    if (addresses.Length > 0)
+                    {
+                        forwardedIp = addresses[0];
+                    }
+                }
+            }
+
+            var ip = request.ServerVariables["REMOTE_ADDR"];
 #else
-            var ip = httpContext.Connection?.RemoteIpAddress;
+            if (CheckForwardedForHeader)
+            {
+                var forwardedHeaders = httpContext.Request.Headers.GetCommaSeparatedValues(ForwardedForHeader);
```

Comment:
Maybe perform `if (Headers.ContainsKey(ForwardedForHeader))` before doing the `GetCommaSeparatedValues`-call. To reduce allocation when nothing to parse.

Reply:
If the header is missing `GetCommaSeparatedValues` returns `StringValues.Empty` which is a static field so I think it shouldn't cause any extra allocation

Reply:
Not when reading the code:

https://github.com/aspnet/HttpAbstractions/blob/master/src/Microsoft.AspNetCore.Http.Abstractions/Extensions/HeaderDictionaryExtensions.cs#L42

Reply:
OK, I will add that check

Reply:
Done